src/Drupal/DrupalExtension/Context/MinkContext.php

```diff
@@ -535,6 +535,23 @@ public function assertNotRegionText($text, $region)
         }
     }
 
+  /**
+   * @Then I should not see ( the text) :text in the :region( region) or no such region
+   *
+   * @throws \Exception
+   *   If region or text within it cannot be found.
+   */
+  public function assertNotRegionTextOrNozSuchRegion($text, $region): void {
+    $regionObj = $this->getSession()->getPage()->find('region', $region);
+    if(!empty($regionObj)) {
+      // Find the text within the region.
+      $regionText = $regionObj->getText();
+      if (strpos($regionText, $text) !== false) {
+        throw new \Exception(sprintf('The text "%s" was found in the region "%s" on the page %s', $text, $region, $this->getSession()->getCurrentUrl()));
+      }
+    }
```

Comment:
@kumikoono Let's use a guard clause:

```suggestion
    $region_obj = $this->getSession()->getPage()->find('region', $region);

    // Find the text within the region.
    $region_text = !empty($region_obj) ? $region_obj->getText() : '';
    if (strpos($region_text, $text) === TRUE) {
      return;
    }

    throw new \Exception(sprintf('The text "%s" was found in the region "%s" on the page %s', $text, $region, $this->getSession()->getCurrentUrl()));
```


```suggestion
    if(!empty($regionObj)) {
      // Find the text within the region.
      $regionText = $regionObj->getText();
      if (strpos($regionText, $text) !== false) {
        throw new \Exception(sprintf('The text "%s" was found in the region "%s" on the page %s', $text, $region, $this->getSession()->getCurrentUrl()));
      }
    }
```